src/main/java/ca/softwarespace/qiyanna/dataaggregator/repositories/sql/SQLRankRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package ca.softwarespace.qiyanna.dataaggregator.repositories.sql;
+
+import static ca.softwarespace.qiyanna.dataaggregator.models.generated.Tables.RANK;
+
+import ca.softwarespace.qiyanna.dataaggregator.models.dto.RankDto;
+import ca.softwarespace.qiyanna.dataaggregator.models.exceptions.RecordNotFoundException;
+import ca.softwarespace.qiyanna.dataaggregator.models.generated.tables.records.RankRecord;
+import ca.softwarespace.qiyanna.dataaggregator.repositories.sql.interfaces.SQLRankRepository;
+import org.jooq.DSLContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class SQLRankRepositoryImpl implements SQLRankRepository {
+  private final DSLContext dsl;
+
+  @Autowired
+  public SQLRankRepositoryImpl(DSLContext dsl) {
+    this.dsl = dsl;
+  }
+
+  @Transactional(readOnly = true)
```

Comment:
Method does to much, return the object to a service layer and let it get transformed by a factory service